api.yaml

```diff
@@ -15,6 +15,130 @@ tags:
     description: Execute, coordinate, or schedule operations
 
 paths:
+  /api/v1/threadpools:
+    get:
+      tags:
+        - visibility
+      summary: Thread pools
+      operationId: threadpools
+      description: |
+        Provides usage statistics of thread pools
+      responses:
+        '200':
+          description: Thread pool statistics
+          content:
+            application/json:
+              schema:
+                type: object
+                additionalProperties:
+                  $ref: '#/components/schemas/ThreadPool'
+                example:
+                  "ReadStage": {
+                    "activeTasks": 0,
+                    "activeTasksLimit": 32,
+                    "pendingTasks": 0,
+                    "completedTasks": 75,
+                    "blockedTasks": 0,
+                    "blockedTasksAllTime": 0
+                  }
+                  "MutationStage": {
+                    "activeTasks": 32,
+                    "activeTasksLimit": 32,
+                    "pendingTasks": 1023,
+                    "completedTasks": 0,
+                    "blockedTasks": 0,
+                    "blockedTasksAllTime": 0
+                  }
+        '503':
+          description: Resource not available due to version conflict or connectivity to cluster
+          content:
+            application/json:
+              schema:
+                type: object
+                items:
+                  $ref: '#/components/schemas/Error'
+
+  /api/v1/settings:
```

Comment:
"v1" indicates major api version, but mentioning it here leaves the possibility of of having multiple major versions of an endpoint where others don't have. Can we use server object to mention the version like shown here: https://swagger.io/specification/#serverObject ?